File: leetcode3/황은지/3875. Construct Uniform Parity Array I.js
/**
 * @param {number[]} nums1
 * @return {boolean}
 */
var uniformArray = function (nums1) {
  //  모두 홀이나 짝이면 차이는 전부 짝 (홀-홀/ 짝-짝=짝)
  // 짝홀이 섞여있다면..?(짝-홀 / 홀-짝=홀)
  //짝홀이 섞여있다면.. 1,2,4,6 => 1,1,3,5 (2개이상이면 다가능) 하나면, 짝-홀 해서 홀조합으로..
  return true;
};

File: leetcode3/최원준/3875. Construct Uniform Parity Array I.py
#

'''
1. 아이디어 :


2. 시간복잡도 :
    O()

3. 자료구조/알고리즘 :


'''

class Solution:
    def uniformArray(self, nums1: list[int]) -> bool:
        n = len(nums1)

        count = 0
        for i in range(n):
            if nums1[i] % 2 == 0:
                count += 1
            else:
                for j in range(n):
                    if i == j:
                        continue
                    if abs(nums1[i] - nums1[j]) % 2 == 0:
                        count+=1
                        break
        
        if count == n:
            return True
        
        count = 0
        for i in range(n):
            if nums1[i] % 2 == 1:
                count += 1
            else:
                for j in range(n):
                    if i == j:
                        continue
                    if abs(nums1[i] - nums1[j]) % 2 == 1:
                        count+=1
                        break

        return count == n
